auth/client/src/main/java/org/wildfly/security/auth/client/EncryptedExpressionConfig.java

```diff
@@ -0,0 +1,245 @@
+     */
+    public EncryptedExpressionConfig addCredentialStore(String credentialStoreName, CredentialStore credentialStore) {
+        Assert.checkNotNullParam("name", credentialStoreName);
+        Assert.checkNotNullParam("credentialStore", credentialStore);
+        Map<String, CredentialStore> credentialStorePair = new HashMap<>();
+        credentialStorePair.put(credentialStoreName, credentialStore);
+        EncryptedExpressionConfig config = new EncryptedExpressionConfig(this, ADD_CREDENTIAL_STORE, credentialStorePair);
+        return config;
+    }
+
+    /**
+     * Create a new configuration which is the same as this configuration, but which uses the given credential stores
+     * instead of the previous ones.
+     *
+     * @param credentialStoreMap the map of the credential store to use in place of the current ones (must not be {@code null})
+     * @return the new configuration
+     */
+    public EncryptedExpressionConfig useCredentialStoreMap(Map<String, CredentialStore> credentialStoreMap) {
+        if (credentialStoreMap == null || credentialStoreMap.isEmpty()) { return this; }
+        return new EncryptedExpressionConfig(this, SET_CREDENTIAL_STORE, credentialStoreMap);
+    }
+
+    /**
+     * Create a new configuration which is the same as this configuration, but removes the given credential store
+     * from the credential store map.
+     *
+     * @param credentialStoreName the name of the credential store to add (must not be {@code null})
+     * @return the new configuration
+     */
+    public EncryptedExpressionConfig removeCredentialStore(String credentialStoreName) {
+        Assert.checkNotNullParam("name", credentialStoreName);
+        EncryptedExpressionConfig config = new EncryptedExpressionConfig(this, REMOVE_CREDENTIAL_STORE, credentialStoreName);
+        return config;
+    }
+
+    /**
+     * Create a new configuration which is the same as this configuration, but which uses the given resolver
+     * in place of existing one.
+     *
+     * @param resolver the Encrypted Expression Resolver to add (must not be {@code null})
+     * @return the new configuration
+     */
+    public EncryptedExpressionConfig addEncryptedExpressionResolver(EncryptedExpressionResolver resolver) {
+        Assert.checkNotNullParam("encrypted expression resolver", resolver);
+        Map<String, EncryptedExpressionResolver.ResolverConfiguration> resolverConfigurationMap = new HashMap<>();
+        resolverConfigurationMap.putAll(resolver.getResolverConfiguration());
+        EncryptedExpressionConfig config = new EncryptedExpressionConfig(this, SET_RESOLVER, resolver);
+        return config;
+    }
+
+    CallbackHandler getCallbackHandler() {
+        return userCallbackHandler;
+    }
+
+    CallbackHandler createCallbackHandler() {
+        return new EncryptedExpressionConfig.ClientCallbackHandler(this);
+    }
+
+    static class ClientCallbackHandler implements CallbackHandler {
```

Comment:
@PrarthonaPaul just a minor, this callback handler is unused so you can remove it

Reply:
removed!